src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -1716,5 +1684,85 @@ private static bool ConvertExceptionMessageFromStringToMessageObject(JObject exc
 
             return modifiedNotification;
         }
+
+        private delegate void ActionOnJObject(JObject jObject);
+
+        private static void ApplyActionOnNodeIfPathExists(string[] possibleNodePaths, JObject rootNode, ActionOnJObject action)
+        {
+            var nodesToUpdate = FindAllNodes(possibleNodePaths, rootNode);            
+
+            foreach (JObject node in nodesToUpdate)
+            {
+                action(node);
+            }
+
+        }
+
+        private static List<JObject> FindAllNodes(string[] fullPaths, JObject rootlog)
```

Comment:
>FindAllNodes [](start = 37, length = 12)

another thought i had was if i should move these up in a helper class - decisions like this make me uncomfortable so i am eager to hear thoughts of more experienced folks here :)

Reply:
great idea! let's wait for a second use.

---
In reply to: [265813845](https://github.com/Microsoft/sarif-sdk/pull/1349#discussion_r265813845) [](ancestors = 265813845)